logstash-core/api/lib/app.rb

```diff
@@ -1,19 +1,32 @@
 # encoding: utf-8
 require "cabin"
 require "logstash/json"
-require "app_helpers"
+require "helpers/app_helpers"
 require "app/service"
 require "app/command_factory"
+require "logstash/util/loggable"
+require "helpers/cabin_compatibility_layer"
 
 module LogStash::Api
   class BaseApp < ::Sinatra::Application
 
     attr_reader :factory
 
+    ::Cabin::Channel.class_eval do
+      include ::Cabin::Mixins::SinatraLogger
+    end
```

Comment:
You don't need to do `class_eval` in that case.

You can do something like this
```
jruby-1.7.23 :001 > class Hello;end
 => nil
jruby-1.7.23 :002 > module Ola; def write;puts "ninja";end;end
 => nil
jruby-1.7.23 :003 > Hello.send(:include, Ola)
 => Hello
jruby-1.7.23 :004 > Ola.write
jruby-1.7.23 :006 > Hello.new.write
ninja
 => nil
```

Or we can do this if you require the exiting file before, the new functionality will be mixed in.

```ruby
# in the helper
require "cabin/namespace"
class Cabin::Channel 
 def write(data, &block)
   self.publish(data, &block)
 end
end
```

Not sure if we have a preferred way